### DISCUSSION TITLE
The next release (6.2.68) and betas (6.2.67b) of Sympa

### DISCUSSION THREAD

**Discussion by ikedas** [sympa-community/sympa#1626] [Release works]
On the next release 6.2.68 and its betas.

**Comment by ikedas** [comment]
For users enabling S/MIME feature and/or X.509 client certificate authentication on web UI:
With sympa-community/sympa#1239, Crypt::OpenSSL::X509 should be updated to 1.909 or later.

**Comment by ikedas** [comment]
To fix a bug in sympa-community/sympa#1235, a new module Archive::Zip::SimpleZip will be introduced.
Archive::Zip may be used as workaround for a while, but Sympa in the near future will drop support for Archive::Zip.

**Comment by xavierba** [comment]
I started to look at getting Archive::Zip::Simple as well as IO::Compress::Zstd and Compress::Stream::Zstd dependencies into Fedora/EPEL.

https://bugzilla.redhat.com/bugzilla/buglist.cgi?bug_id=2011830&bug_id_type=anddependson&format=tvp&list_id=12181677#

edit: add link to review request

**Comment by xavierba** [comment]
IO::Compress::Base which is needed by Archive::Zip::SImpleZip is in base RHEL so I will need to work against fixed versions, that is:
EL7:  IO::Compress::Base 2.061 and thus at most Archive::Zip::SImpleZip 0.010
EL8:  IO::Compress::Base 2.081 and thus at most Archive::Zip::SImpleZip 0.021

Is that OK or is a minimum version of Archive::Zip::SimpleZip enforced ?

**Comment by ikedas** [comment]
According to my limited testing, another module Archive::Zip::Simple*Un*zip included in Archive-Zip-SimpleZip needs to be version 0.024 or later which requires IO::Compress::Base 2.081. (in general the latest version is more desirable, though).

For the environments where the latest package is not available, a fallback to Archive-Zip is left.

**Comment by ikedas** [comment]
I think the first beta, 6.2.67b.1, would be out in this weekend to the next week.

**Comment by ikedas** [comment]
Done!

**Comment by ikedas** [comment]
### Careful checks are wanted

On the next beta (6.2.67b.2), there are some big changes mainly due to refactoring work.  They include things below.  When the b.2 will be released, please install it and check if related functions work properly.

  - `--add` and `--del` options were added on sympa.pl, and `add` and `del` request handlers became possible to add/del owners/editors too.
    See sympa-community/sympa#911, sympa-community/sympa#1272, sympa-community/sympa#1251 and related issues.

  - Refactoring the code parsing `auth.conf` and processing CAS and generic SSO authentications.
    See sympa-community/sympa#1255

  - Refactoring on `wwsympa.fcgi`
    See sympa-community/sympa#1247

**Comment by ikedas** [comment]
In my personal thought beta 2 (6.2.67b.2) will be released around 25th Nov.

**Comment by xavierba** [comment]
Archive::Zip::SimpleZip is on its way to Fedora and EPEL 8. Fedora will get latest 0.40, while EL8 will have 0.25.

**Comment by ikedas** [comment]
I think it would be preferable to release one more beta.
Proposed release date of 6.2.67b.3 is around Dec. 21st, and the date of 6.2.68 would be two weeks after, '22 Jan. 4th.

**Comment by ikedas** [comment]
Done!

**Comment by racke** [comment]
Thanks!

**Comment by ikedas** [comment]
I may be out for early next week.  Release of 6.2.68 would be postponed to the next weekend.

**Comment by ikedas** [comment]
@dpc22 will check some PRs. And, when he's done, I'll release 6.2.68.
Any issues/PRs submitted after now will not be marked as milestone 6.2.68: They will be postponed to the next release.

**6.2.68 will be released around 21st or 22nd Jan**.

**Comment by ikedas** [comment]
**Done!**

**Comment by xavierba** [comment]
Sympa 6.2.68 has been built in Fedora Rawhide and I'm going to build it next for Fedora 35, 34 and EL 7.

Sympa for EL8 still live in my COPR repo, as I wanted to complete the work on switching to systemd sockets/multiwatch before making it available in EPEL. As @racke worked on a similar topic for Debian, it might be a good time to continue the discussion at https://github.com/sympa-community/sympa/issues/799 and plan to get the modified systemd unit files in upstream sympa.

Also, sympa for EPEL 9 is almost ready, I've been actively working on getting all the needed perl modules in EPEL 9 since start of December. It's only missing a couple perl modules for mhonarc. And indeed, spawn-fcgi if not switched to systemd socket/multiwatch.
